### ISSUE TITLE
require 'binary_search/native' doesn't work on OS X

### ISSUE THREAD

**Issue by ghazel** [tyler/binary_search#2]
```
$ sudo gem install binary_search
Building native extensions.  This could take a while...
Successfully installed binary_search-0.2.0
1 gem installed
$ irb
>> require 'binary_search/native'
LoadError: no such file to load -- /Library/Ruby/Gems/1.8/gems/binary_search-0.2.0/lib/binary_search/../../ext/binary_search
    from /System/Library/Frameworks/Ruby.framework/Versions/1.8/usr/lib/ruby/1.8/rubygems/custom_require.rb:31:in `gem_original_require'
    from /System/Library/Frameworks/Ruby.framework/Versions/1.8/usr/lib/ruby/1.8/rubygems/custom_require.rb:31:in `require'
    from /Library/Ruby/Gems/1.8/gems/binary_search-0.2.0/lib/binary_search/native.rb:1
    from /System/Library/Frameworks/Ruby.framework/Versions/1.8/usr/lib/ruby/1.8/rubygems/custom_require.rb:36:in `gem_original_require'
    from /System/Library/Frameworks/Ruby.framework/Versions/1.8/usr/lib/ruby/1.8/rubygems/custom_require.rb:36:in `require'
    from (irb):1
>> require 'binary_search/pure'
=> true
```

**Comment by al** [comment]
The gem in rubygems is broken (missing the C source file). You can try cloning from the repo (`gem 'binary_search', git: 'git://github.com/tyler/binary_search.git'`) but there are problems compiling:

```
Gem::Installer::ExtensionBuildError: ERROR: Failed to build gem native extension.

        /Users/alan/.rvm/rubies/ruby-1.9.3-p194/bin/ruby extconf.rb 
creating Makefile

make
compiling binary_search.c
binary_search.c:7:29: error: no member named 'len' in 'struct RArray'
  int upper = RARRAY(self)->len - 1;
              ~~~~~~~~~~~~  ^
binary_search.c:12:63: error: no member named 'ptr' in 'struct RArray'
    comp = FIX2INT(rb_funcall(value, id_cmp, 1, RARRAY(self)->ptr[i]));
           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~
/Users/alan/.rvm/rubies/ruby-1.9.3-p194/include/ruby-1.9.1/ruby/ruby.h:519:45: note: expanded from macro 'FIX2INT'
#define FIX2INT(x) ((int)rb_fix2int((VALUE)(x)))
                                            ^
2 errors generated.
make: *** [binary_search.o] Error 1
```

Presumably Ruby 1.8.7 is required?

**Comment by tmm1** [comment]
This should be fixed in 0.3.0
